### PR TITLE
CTest & script running flow and compareSummary

### PR DESCRIPTION
**Pull request by kristfho**
Added a test that uses the compareSummary binary from my previous [PR](https://github.com/OPM/opm-output/pull/56). It runs only on the SPE9_CP-case, however this can easily be changed.

### CODE REVIEW

File: TestSummaryFiles.cmake [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# TEST: compareSummary 
+###########################################################################
+
+# Input: 
+#   - casename: basename (no extension) 
+#
+macro (add_test_compareSummary casename filename)
+  
+  # Add test that runs upscale_perm and outputs the results to file
+  opm_add_test(comapreSummary_${filename} NO_COMPILE
```
Comment (akva2) [inline TestSummaryFiles.cmake]: compare.

File: jenkins/run-compareSummary.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
```
Comment (akva2) [inline jenkins/run-compareSummary.sh]: this file is not a jenkins script so wrong path. i suggest putting it in tests/

File: CMakeLists.txt [hunk 3]
```diff
@@ -156,4 +156,8 @@ if (HAVE_OPM_DATA)
         opm_add_python_test( check_INIT_NORNE ${PROJECT_SOURCE_DIR}/tests/compare_INIT.py $<TARGET_FILE:flow> ${OPM_DATA_ROOT}/norne/NORNE_ATW2013.DATA ${OPM_DATA_ROOT}/norne/ECL.2014.2/NORNE_ATW2013.INIT PORO PORV PERMX )
     endif()
 
+		if(HAVE_OPM_OUTPUT)
```
Comment (akva2) [inline CMakeLists.txt]: indent. use spaces please.

File: jenkins/run-compareSummary.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+EXE_NAME="$8"
+TEST_ARGS="${@:9:100}"
+
+cd ${RESULT_PATH}
+${BINPATH}/${EXE_NAME} ${TEST_ARGS}
+cd ..
+
+#~/opm-output/build/bin/compareSummary ${RESULT_PATH}/${FILENAME} ${INPUT_DATA_PATH}/${CASENAME}/opm-simulation-reference/${FILENAME}
+
+${OPM_OUTPUT_ROOT}/build/bin/compareSummary ${RESULT_PATH}/${FILENAME} ${INPUT_DATA_PATH}/${CASENAME}/opm-simulation-reference/${FILENAME}
```
Comment (akva2) [inline jenkins/run-compareSummary.sh]: this won't work. there's nothing mandating the use of a sibling-build for opm-output. see the patchi sent earlier this week  (or was it a branch link, don't recall)  where we set the path to this binary as a cmake variable in Findopm-output.cmake.

File: CMakeLists.txt (hunk 3 above)
Comment (joakim-hove) [inline CMakeLists.txt]: Currently `opm-output`  is a hard dependency for `opm-simulators`, and I guess there are no plans to change that - so I would suggest dropping the `if (HAVE_OPM_OUTPUT)`  test completely.

File: TestSummaryFiles.cmake [hunk 5]
```diff
@@ -0,0 +1,44 @@
+# Set absolute tolerance to be used for testing
+set(tol 1e-2)
+
+# Define some paths
+set(RESULT_PATH ${PROJECT_BINARY_DIR}/tests/results)
```
Comment (joakim-hove) [inline TestSummaryFiles.cmake]: I don't know how well it fits in with this model - but I have used the [TestArea](https://github.com/Ensembles/ert/blob/master/devel/libert_utilxx/src/TestArea.cpp) quite a lot - and find it very convenient. The `TestArea`class works like this:

1. Create a random temporary directory and do a `chdir( )` there.
2. Optionally you can use utility functions which are aware of your current cwd and the original cwd to conveniently copy files and directories.
3. When it goes out of scope the directory and the files are removed[1,2]

Example usage:
```C++
#include <ert/util/TestArea.hpp>
...
{
    ERT::TestArea ta("my_funny_test");
    // Now cwd is in some random directory().
    ta.copyFile( "path/relative/to/original/cwd");
    // Do your stuff ..
}
// Now the original CWD has been recovered, and the temporary directory has been wiped.
```


[1]:  There will be a directory stem from `/tmp/$USER/...` left behind.
[2]:  For debugging purposes you can set the `store`flag to `true` - in which case the directory will be left intact.

File: TestSummaryFiles.cmake [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+###########################################################################
+# TEST: compareSummary 
+###########################################################################
+
+# Input: 
+#   - casename: basename (no extension) 
+#
+macro (add_test_compareSummary casename filename)
+  
+  # Add test that runs upscale_perm and outputs the results to file
```
Comment (akva2) [inline TestSummaryFiles.cmake]: really?

**Comment by akva2** [comment]
with the minor addressed, i think this is ready for merge. squashed of course.

File: TestSummaryFiles.cmake [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  add_custom_target(test-suite)
+endif()
+
+add_custom_target(clear_test_data ${CMAKE_COMMAND} -E remove_directory ${RESULT_PATH}
+                  COMMAND ${CMAKE_COMMAND} -E make_directory ${RESULT_PATH}
+                  COMMENT "Removing old test results")
+
+add_dependencies(check clear_test_data)
+
+opm_set_test_driver(${PROJECT_SOURCE_DIR}/tests/run-compareSummary.sh "")
```
Comment (akva2) [inline TestSummaryFiles.cmake]: this path has changed.

File: tests/run-regressionTest.sh [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+INPUT_DATA_PATH="$1"
+RESULT_PATH="$2"
+BINPATH="$3"
+FILENAME="$4"
+ABS_TOL="$5"
+REL_TOL="$6"
+SUMMARY_REGRESSION_TEST_COMMAND="$7"
+RESTART_REGRESSION_TEST_COMMAND="$8"
+INIT_REGRESSION_TEST_COMMAND="$9"
+EXE_NAME="$10"
```
Comment (akva2) [inline tests/run-regressionTest.sh]: when going above 9, you have to use ${10} or it evaluates as ${1}0 - first param with a 0 at the end.

File: TestSummaryFiles.cmake [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+macro (add_test_compareSummary casename filename)
+  
+  # Add test that runs upscale_perm and outputs the results to file
+  opm_add_test(compareSummary_${filename} NO_COMPILE
+               EXE_NAME flow   
+	       DRIVER_ARGS ${OPM_DATA_ROOT}/${casename} ${RESULT_PATH}
+                           ${CMAKE_BINARY_DIR}/bin
+                           ${filename}
+                           ${abs_tol} ${rel_tol}
+			   ${SUMMARY_REGRESSION_TEST_COMMAND}
```
Comment (akva2) [inline TestSummaryFiles.cmake]: ${OPM_COMPARE_SUMMARY_COMMAND} etc.

**Comment by akva2** [comment]
with all aforementioned fixes the successfully executed the regression test. it fails though as the reference data is already out of date (more data in summary file now it seems, likely some addition?)

**Comment by akva2** [comment]
jenkins build this opm-common=158 opm-output=55 please

**Comment by akva2** [comment]
jenkins build this opm-common=158 opm-output=55 please

File: tests/run-regressionTest.sh [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+RESULT_PATH="$2"
+BINPATH="$3"
+FILENAME="$4"
+ABS_TOL="$5"
+REL_TOL="$6"
+SUMMARY_REGRESSION_TEST_COMMAND="$7"
+RESTART_REGRESSION_TEST_COMMAND="$8"
+INIT_REGRESSION_TEST_COMMAND="$9"
+EXE_NAME="${10}"
+TEST_ARGS="${@:{11}:100}"
```
Comment (akva2) [inline tests/run-regressionTest.sh]: not this, was only the one above that needs parsing help (ie only when dereferencing the variable).

Comment (bska) [inline tests/run-regressionTest.sh]: Is there a reason we're not using
```sh
shift 10
TEXT_ARGS="$@"
```
or for that matter passing `"$@"` directly to the comparator?  That would be the traditional approach and we wouldn't need to invent an arbitrary upper bound on the number of positional parameters.

Comment (akva2) [inline tests/run-regressionTest.sh]: nope. no reason, good point.

**Comment by akva2** [comment]
jenkins build this opm-common=158 opm-output=55 please

**Comment by akva2** [comment]
jenkins build this opm-common=158 opm-output=55 opm-data=118 please

File: CMakeLists.txt [hunk 11]
```diff
@@ -156,4 +156,10 @@ if (HAVE_OPM_DATA)
         opm_add_python_test( check_INIT_NORNE ${PROJECT_SOURCE_DIR}/tests/compare_INIT.py $<TARGET_FILE:flow> ${OPM_DATA_ROOT}/norne/NORNE_ATW2013.DATA ${OPM_DATA_ROOT}/norne/ECL.2014.2/NORNE_ATW2013.INIT PORO PORV PERMX )
     endif()
 
+    if(HAVE_OPM_OUTPUT)
```
Comment (joakim-hove) [inline CMakeLists.txt]: You *have* opm-output; this test is purely noise.

**Comment by akva2** [comment]
jenkins build this opm-common=158 opm-output=55 please

**Pull request merged**